src/client/datascience/raw-kernel/rawJupyterSession.ts

```diff
@@ -86,50 +85,41 @@ export class RawJupyterSession extends BaseJupyterSession {
-                    createPromiseFromCancellation({
-                        cancelAction: 'reject',
-                        defaultValue: new CancellationError(),
-                        token: cancelToken
-                    })
-                ]),
-                timeout
-            );
+            newSession = await this.startRawSession(kernelConnection, timeout, cancelToken);
 
             // Only connect our session if we didn't cancel or timeout
-            if (newSession instanceof CancellationError) {
+            sendTelemetryEvent(Telemetry.RawKernelSessionStartSuccess);
+            traceInfo('Raw session started and connected');
+            this.setSession(newSession);
+
+            // Listen for session status changes
+            this.session?.statusChanged.connect(this.statusHandler); // NOSONAR
+
+            // Update kernelspec and interpreter
+            this.kernelConnectionMetadata = newSession.kernelProcess?.kernelConnectionMetadata;
+
+            this.outputChannel.appendLine(
+                localize.DataScience.kernelStarted().format(
+                    getDisplayNameOrNameOfKernelConnection(this.kernelConnectionMetadata)
+                )
+            );
+        } catch (error) {
+            this.connected = false;
+            if (error instanceof CancellationError) {
                 sendTelemetryEvent(Telemetry.RawKernelSessionStartUserCancel);
                 traceInfo('Starting of raw session cancelled by user');
-                throw newSession;
-            } else if (newSession === null) {
+                throw error;
+            } else if (error instanceof TimedOutError) {
                 sendTelemetryEvent(Telemetry.RawKernelSessionStartTimeout);
                 traceError('Raw session failed to start in given timeout');
+                // Translate into original error
                 throw new RawKernelSessionStartError(kernelConnection);
             } else {
-                sendTelemetryEvent(Telemetry.RawKernelSessionStartSuccess);
-                traceInfo('Raw session started and connected');
-                this.setSession(newSession);
-
-                // Listen for session status changes
-                this.session?.statusChanged.connect(this.statusHandler); // NOSONAR
-
-                // Update kernelspec and interpreter
-                this.kernelConnectionMetadata = newSession.kernelProcess?.kernelConnectionMetadata;
-
-                this.outputChannel.appendLine(
-                    localize.DataScience.kernelStarted().format(
-                        getDisplayNameOrNameOfKernelConnection(this.kernelConnectionMetadata)
-                    )
-                );
+                // Send our telemetry event with the error included
+                sendTelemetryEvent(Telemetry.RawKernelSessionStartException, undefined, undefined, error);
+                traceError(`Failed to connect raw kernel session: ${error}`);
```

Comment:
I think having `IPYkernelNotInstalled` error or the like is useful here.
Basically user cancelled it (see lines 109 & 110)
We know user cancelled it. & not installing ipykernel is the same, meaning user has cancelled the start of the kernel.

Reply:
Else we log the wrong telemetry & wrong message.
We could optionally log telemetry indicating user opted not to install hence startup failed.

Reply:
Should be fixed now.